com.microsoft.mrtk.uxcore/Keyboard/NonNative/NonNativeKeyboardShiftEvent.cs

```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using UnityEngine.Events;
+
+namespace Microsoft.MixedReality.Toolkit.UX
+{
+    [Serializable]
+    public class NonNativeKeyboardShiftEvent : UnityEvent<bool> { }
```

Comment:
XRI does this neat thing where instead of explicitly typing their UnityEvents like `UnityEvent<bool>`, they build `EventArgs` wrapper classes that allow for extensibility of the event args passed through. I.e., later on, they can decide to pass additional data through the event while not breaking existing subscribers to that event.

```
    /// <summary>
    /// <see cref="UnityEvent"/> that Unity invokes when an Interactor first initiates hovering over an Interactable.
    /// </summary>
    [Serializable]
    public sealed class HoverEnterEvent : UnityEvent<HoverEnterEventArgs>
    {
    }

    /// <summary>
    /// Event data associated with the event when an Interactor first initiates hovering over an Interactable.
    /// </summary>
    public class HoverEnterEventArgs : BaseInteractionEventArgs
    {
        /// <summary>
        /// The Interactor associated with the interaction event.
        /// </summary>
        public new IXRHoverInteractor interactorObject
        {
            get => (IXRHoverInteractor)base.interactorObject;
            set => base.interactorObject = value;
        }

        /// <summary>
        /// The Interactable associated with the interaction event.
        /// </summary>
        public new IXRHoverInteractable interactableObject
        {
            get => (IXRHoverInteractable)base.interactableObject;
            set => base.interactableObject = value;
        }

        /// <summary>
        /// The Interaction Manager associated with the interaction event.
        /// </summary>
        public XRInteractionManager manager { get; set; }
    }
```

Reply:
I thought about this and also talked to @keveleigh. Ultimately, I think it comes down to whether the event wrappers are expected to be changed or not. I think given our case here the wrappers around bool, string and the key should be all we need, and I don't really anticipate an expansion there. So I believe we are good even if we don't go down this event args route.

Reply:
Alright, sounds good to me.